libshpool/src/daemon/prompt.rs

```diff
@@ -67,12 +67,8 @@ pub fn maybe_inject_prefix(
                SHPOOL__OLD_PROMPT_COMMAND="${{PROMPT_COMMAND}}"
                SHPOOL__OLD_PS1="${{PS1}}"
                function __shpool__prompt_command() {{
-                  PS1="${{SHPOOL__OLD_PS1}}"
-                  for prompt_hook in ${{SHPOOL__OLD_PROMPT_COMMAND}}
-                  do
-                    ${{prompt_hook}}
-                  done
-                  PS1="{prompt_prefix}${{PS1}}"
+                  eval "${{SHPOOL__OLD_PROMPT_COMMAND}}"
+                  PS1="{prompt_prefix}${{SHPOOL__OLD_PS1}}"
```

Comment:
This will clobber any effect that the old prompt command had on PS1 because it sets PS1 to the old PS1 value. If the prompt hook we are extending modifies PS1 in any way, that will no longer get picked up. This line should be

```
PS1="{prompt_prefix}${{PS1}}"
```

to fix that.

Reply:
Yeah, that looks like it will work:

```
$ arr=('echo hi' 'echo bye' 'echo blip bloop blop')
$ echo $(IFS="; "; echo "${arr[*]}")
echo hi;echo bye;echo blip bloop blop
$ eval "$(IFS="; "; echo "${arr[*]}")"
hi
bye
blip bloop blop
```